File: tests/fixtures/Sample.java
package com.example.auth;

import com.example.models.User;
import com.example.db.Database;
import java.util.Optional;

/**
 * Authentication service
 */
public class AuthService {

    private static final int MAX_RETRIES = 5;
    private final Database db;

    public AuthService(Database db) {
        this.db = db;
    }

    /**
     * Authenticate user with credentials
     */
    public Optional<User> login(String email, String password) {
        User user = db.findByEmail(email);
        if (user != null && verifyPassword(user, password)) {
            return Optional.of(user);
        }
        return Optional.empty();
    }

    private boolean verifyPassword(User user, String password) {
        return !password.isEmpty();
    }
}

public interface Authenticator {
    Optional<User> authenticate(String email, String password);
}

public enum Role {
    ADMIN,
    MEMBER
}

File: tests/fixtures/sample.rs
use std::collections::HashMap;
use crate::db::Database;
use super::utils::validate;

/// Maximum retry count
pub const MAX_RETRIES: u32 = 5;

/// Represents a user in the system
pub struct User {
    pub id: String,
    pub email: String,
    role: Role,
}

pub enum Role {
    Admin,
    Member,
}

/// Authentication trait
pub trait Authenticator {
    fn authenticate(&self, email: &str, password: &str) -> Result<User, AuthError>;
}

pub type AuthResult = Result<User, AuthError>;

impl User {
    pub fn new(id: String, email: String) -> Self {
        User { id, email, role: Role::Member }
    }

    pub fn is_admin(&self) -> bool {
        matches!(self.role, Role::Admin)
    }
}

/// Authentication service
pub struct AuthService {
    db: Database,
}

impl AuthService {
    pub fn new(db: Database) -> Self {
        AuthService { db }
    }

    /// Authenticate user with credentials
    pub fn login(&self, email: &str, password: &str) -> AuthResult {
        let user = self.db.find_by_email(email)?;
        if verify_password(&user, password) {
            Ok(user)
        } else {
            Err(AuthError::InvalidPassword)
        }
    }
}

fn verify_password(user: &User, password: &str) -> bool {
    !password.is_empty()
}

pub struct AuthError;

impl AuthError {
    pub const InvalidPassword: AuthError = AuthError;
}
